rfcs/053-lambda-logging/README.md

```diff
@@ -0,0 +1,63 @@
+
+If we built our own Lambda extension, we would expose a Terraform module for provisioning Lambdas that included the extension and necessary config, IAM roles etc etc. We could write it in a language of our choice and use it with any native (ie non-Docker lambda). 
+
+It could potentially use fluent bit for streaming the logs, instead of implementing that manually, although it looks like [others](https://github.com/aws-samples/aws-lambda-extensions/pull/42) have struggled with losing logs for short-lived invocations (probably due to buffering, and possibly resolveable with use of Lambda lifecycle hooks).
+
+**There is a caveat with this option** - it doesn't work transparently for containerised (ie non-native) Lambdas. Instead, the packaged extension has to be [added to the container image](https://docs.aws.amazon.com/lambda/latest/dg/extensions-configuration.html). 
+
+We use containerised Lambdas in the concepts pipeline: we would either have to switch to using native Java  runtimes, or provide our own base images which include the extension code.
+
+There would be no additional cost associated with using a Lambda extension, other than the necessary network egress.
+
+![Native lambda extensions architecture](https://user-images.githubusercontent.com/4429247/203977301-cb6c2cdb-685e-4b62-9ef1-5396fc349b12.png)
+
+## A pragmatic path forward
+
+*Option (1) does not preclude option (2) as further work*: we would already be providing a Terraform module for consumers, and if the Elastic Serverless Forwarder proved insufficient, we would already have written a module to stream the correct format of logs to Elasticsearch. 
+
+This fact clarifies that option (1) - streaming of logs from CloudWatch - is the better starting point. If it proves insufficient or impractical, the work we will have done already will mostly still be useful for option (2).
+
+The work plan might look something like this:
+
+- Provision an Elastic serverless forwarder
+- Create a log group subscription and verify that the forwarder works
+- Create a Terraform module for Lambdas with log groups that subscribe to the forwarder
+- Roll this out across our Lambdas
+
+## Caveats and questions
+
+- We've seen CloudWatch logs for Lambdas go missing before - is this a risk? We believe this is due to (mis)configuration of log retention periods, but we're not sure.
+- Should we use Kinesis or not? Some reasons for using it are given above: I think it would be a good idea.
+- What index should Lambda logs go into? The ECS logs go into `firelens-<yyyy.mm.dd>` indices - we probably shouldn't do the same for these, as they aren't using firelens and may have a different schema.
```

Comment:
I don't mind, but I don't think it should include the word "Lambda".

Yesterday you mentioned using some sort of Elastic common schema, which isn't mentioned above. If we do use a properly generic schema that we might use for other, non-Lambda applications in the future, let's not box ourselves into a corner with naming.

Reply:
Yeah, I would rather the ECS one wasn't firelens tbh!